aiohttp/connector.py

```diff
@@ -964,6 +971,18 @@ async def _wrap_create_connection(
             async with ceil_timeout(
                 timeout.sock_connect, ceil_threshold=timeout.ceil_threshold
             ):
+                if self._network_interface:
+                    kwargs["sock"] = socket.socket(socket.AF_INET, socket.SOCK_STREAM)
+                    kwargs["sock"].setsockopt(
+                        socket.SOL_SOCKET,
+                        socket.SO_BINDTODEVICE,
+                        self._network_interface.encode(),
+                    )
+                    if self._local_addr:
+                        kwargs["sock"].bind(self._local_addr)
+                    kwargs["sock"].setblocking(False)
+                    kwargs["sock"].connect_ex((args[-2], args[-1]))
+                    args = args[:-2]  # only send socket instead host/ip
```

Comment:
This feels like sketchy hacking. At least, use a named var to split this like
```python
*args, sock_hostname, sock_port = args
```
and use pass those vars to the socket initialization helper.

Reply:
Looks cleaner agreed. I was unsure what the overall coding style was for aiohttp, so i tried to make code that's uses atleast resources / inexpensive as possible.